src/routes/Register.js

```diff
@@ -0,0 +1,121 @@
+      password: "",
+      confirmPassword: "",
+    },
+    validationSchema: registerValidationSchema,
+    onSubmit: (values) => {
+      // update next line to add values to database
+      console.log(values);
+    },
+  });
+  const classes = formStyles();
+  return (
+    <div className={classes.main}>
+      <Container maxWidth="sm">
+        <Card className={classes.card}>
+          <Typography component="h1" variant="h5" align="center">
+            Register
+          </Typography>
+          <form onSubmit={formik.handleSubmit}>
+            <TextField
+              id="name"
+              name="name"
+              label="Name"
+              value={formik.values.name}
+              onChange={formik.handleChange}
+              error={formik.touched.name && Boolean(formik.errors.name)}
+              helperText={formik.touched.name && formik.errors.name}
+              className={classes.fields}
+            />
+            <TextField
+              id="email"
+              name="email"
+              label="Email"
+              type="email"
+              value={formik.values.email}
+              onChange={formik.handleChange}
+              error={formik.touched.email && Boolean(formik.errors.email)}
+              helperText={formik.touched.email && formik.errors.email}
+              className={classes.fields}
+            />
+            <TextField
+              id="confirmEmail"
+              name="confirmEmail"
+              label="Confirm Email"
+              type="email"
+              value={formik.values.confirmEmail}
+              onChange={formik.handleChange}
+              error={
+                formik.touched.confirmEmail &&
+                Boolean(formik.errors.confirmEmail)
+              }
+              helperText={
+                formik.touched.confirmEmail && formik.errors.confirmEmail
+              }
+              className={classes.fields}
+            />
+            <TextField
+              id="password"
+              name="password"
+              label="Password"
+              type="password"
+              value={formik.values.password}
+              onChange={formik.handleChange}
+              error={formik.touched.password && Boolean(formik.errors.password)}
+              helperText={formik.touched.password && formik.errors.password}
+              className={classes.fields}
+            />
+            <TextField
+              id="confirmPassword"
+              name="confirmPassword"
+              label="Confirm Password"
+              type="password"
+              value={formik.values.confirmPassword}
+              onChange={formik.handleChange}
+              error={
+                formik.touched.confirmPassword &&
+                Boolean(formik.errors.confirmPassword)
+              }
+              helperText={
+                formik.touched.confirmPassword && formik.errors.confirmPassword
+              }
+              className={classes.fields}
+            />
```

Comment:
Just a thought to try out and see what you want to do, but since almost all of the text fields have the same props it might be worth creating a custom component that maps through a list of fieldNames `[name, email, password, ...etc]` and then generates this list dynamically. You can handle all error cases in the validation function that gets run before the submission process of the handle submit. Try it out and let me know what you think. Totally cool if you don't like it.

Reply:
We'd have to have an array of objects:
```
{ id, name, label, value, error, helperText, etc....... } 
```
for each field. It would save us **a little code** but not a ton. This treads the line of too repetitive, I think it is fine for now.